packages/salesforcedx-lightning-lsp-common/src/types/fileSystemTypes.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * File system data types for LSP communication
+ * These types represent file system operations that the client will perform
+ * and send data to the language server instead of the server accessing files directly
+ */
+
+export interface FileSystemData {
+    /** File contents keyed by URI */
+    fileContents: Map<string, string>;
+    /** Directory listings keyed by URI */
+    directoryListings: Map<string, DirectoryEntry[]>;
+    /** File stats keyed by URI */
+    fileStats: Map<string, FileStat>;
+    /** Workspace configuration data */
+    workspaceConfig: WorkspaceConfig;
+}
+
+export interface DirectoryEntry {
+    name: string;
+    type: 'file' | 'directory';
+    uri: string;
+}
+
+export interface FileStat {
+    type: 'file' | 'directory';
+    ctime: number;
+    mtime: number;
+    size: number;
+    exists: boolean;
+}
+
+export interface WorkspaceConfig {
+    /** Workspace root URI */
+    rootUri: string;
+    /** Workspace type (SFDX, CORE_ALL, etc.) */
+    workspaceType: string;
+    /** Package directories for SFDX projects */
+    packageDirectories?: string[];
+    /** TypeScript configuration */
+    tsConfig?: any;
+    /** JavaScript configuration */
+    jsConfig?: any;
+    /** Force ignore patterns */
+    forceIgnore?: string[];
+}
+
+/**
+ * LSP notification types for file system operations
+ */
+export namespace FileSystemNotifications {
```

Comment:
let's avoid namespaces.  If you really like the feel of `foo.bar` use an object {}